gulpfile.js

```diff
@@ -1,74 +1,83 @@
-var name = require("./package.json").moduleName,
-	fs = require("fs"),
-	gulp = require("gulp"),
-	plugins = require("gulp-load-plugins")();
-sass = require("gulp-sass")(require("sass"));
+"use strict";
 
-var head = fs.readFileSync("./node_modules/@electerious/modulizer/head.js", {
-		encoding: "utf8",
-	}),
-	foot = fs.readFileSync("./node_modules/@electerious/modulizer/foot.js", {
-		encoding: "utf8",
-	});
+const name = require("./package.json").moduleName;
+const gulp = require("gulp");
+const browserify = require("browserify");
+const babelify = require("babelify");
+const source = require("vinyl-source-stream");
+const buffer = require("vinyl-buffer");
+const uglify = require("gulp-uglify");
+const sass = require("gulp-sass")(require("sass"));
+const rename = require("gulp-rename");
+const autoprefixer = require("gulp-autoprefixer");
+const csso = require("gulp-csso");
 
-var catchError = function (err) {
+const catchError = function (err) {
 	console.log(err.toString());
 	this.emit("end");
 };
 
-gulp.task("styles", function () {
-	gulp
-		.src("./src/styles/main.scss")
-		.pipe(sass().on("error", catchError))
+gulp.task("scripts", function () {
+	return browserify({
+		entries: "./src/scripts/basicContext.js",
+		standalone: name,
+	})
+		.transform(
+			babelify.configure({
+				presets: ["@babel/preset-env"],
+			})
+		)
+		.bundle()
+		.on("error", catchError)
+		.pipe(source(name + ".min.js"))
+		.pipe(buffer())
+		.pipe(uglify())
 		.on("error", catchError)
-		.pipe(plugins.concat(name + ".min.css", { newLine: "\n" }))
-		.pipe(plugins.autoprefixer("last 2 version", "> 1%"))
-		.pipe(plugins.minifyCss())
 		.pipe(gulp.dest("./dist"));
+});
 
-	gulp
-		.src("./src/styles/themes/*.scss")
-		.pipe(sass().on("error", catchError))
+gulp.task("styles", function () {
+	return gulp
+		.src("./src/styles/main.scss")
+		.pipe(sass())
 		.on("error", catchError)
-		.pipe(
-			plugins.rename(function (path) {
-				path.basename += ".min";
-			})
-		)
-		.pipe(plugins.autoprefixer("last 2 version", "> 1%"))
-		.pipe(plugins.minifyCss())
-		.pipe(gulp.dest("./dist/themes"));
+		.pipe(rename((path) => (path.basename = name + ".min")))
+		.pipe(autoprefixer("last 2 version", "> 1%"))
+		.pipe(csso({ restructure: false }))
+		.pipe(gulp.dest("./dist"));
+});
 
+gulp.task("addons", function () {
 	return gulp
 		.src("./src/styles/addons/*.scss")
-		.pipe(sass().on("error", catchError))
+		.pipe(sass())
 		.on("error", catchError)
 		.pipe(
-			plugins.rename(function (path) {
-				path.basename += ".min";
+			rename((path) => {
+				path.basename = path.basename + ".min";
```

Comment:
Just a small visual improvement. Twelve lines above we have the same code but with normal parentheses `()` instead of the curly braces `{}`. As the anonymous inline function has only a single line, the curly braces are not really required. This saves some line breaks. Above we only have

```js
pipe(rename((path) => (path.basename = name + ".min")))
```

Yes, this is real nitpicking, but I like consistency. ;) Same holds 12 lines below.